pandas/core/dtypes/astype.py

```diff
@@ -246,3 +247,39 @@ def astype_array_safe(
             raise
 
     return new_values
+
+
+def astype_is_view(dtype: DtypeObj, new_dtype: DtypeObj) -> bool:
+    """Checks if astype avoided copying the data.
+
+    Parameters
+    ----------
+    dtype : Original dtype
+    new_dtype : target dtype
+
+    Returns
+    -------
+    True if new data is a view, False otherwise
+    """
+    if dtype == new_dtype:
+        return True
+
+    elif isinstance(dtype, np.dtype) and isinstance(new_dtype, np.dtype):
+        # Only equal numpy dtypes avoid a copy
+        return False
+
+    if is_string_dtype(dtype) and is_string_dtype(new_dtype):
+        return True
+
+    elif is_string_dtype(dtype) or is_string_dtype(new_dtype):
+        return False
+
+    elif dtype.kind in "mM" and new_dtype.kind in "mM":
+        return True
+
+    elif getattr(dtype, "numpy_dtype", dtype) == getattr(
+        new_dtype, "numpy_dtype", new_dtype
```

Comment:
This is for our masked arrays? If so, can you add some comments explaining that?

Reply:
pyarrow as well, both don't copy